demos/paywall_bot/demo.py

```diff
@@ -0,0 +1,141 @@
+"""The module is a script for running the bot."""
+
+# ruff: noqa: I001
+
+from hammett.conf import settings
+from hammett.core import Application, Button, Screen
+from hammett.core.constants import DEFAULT_STATE, SourceTypes
+from hammett.core.handlers import register_button_handler
+from hammett.core.mixins import StartMixin
+from hammett.core.permission import ignore_permissions
+from hammett.core.persistence import RedisPersistence
+
+from permissions import PaywallPermission
+
+MAIN_MENU_SCREEN_DESCRIPTION = (
+    "Congratulations! You've successfully made a fake payment. Now you can see the "
+    "<b>Main Menu</b> screen even after typing the /start command. "
+    "\n"
+    "<i>If you'd like to try the demo again, just click the button below "
+    "to make a fake refund 👇</i>"
+)
+
+FAKE_PAYMENT_SCREEN_DESCRIPTION = (
+    'To continue using the bot, you need to make a <i>fake</i> payment.\n'
+    '\n'
+    'Do you want to continue?'
+)
+
+PAYMENT_SCREEN_DESCRIPTION = (
+    "Welcome to HammettPaywallBot!\n"
+    "\n"
+    "Now you see the <b>Payment</b> screen, and you <i>won't see any other screens</i> until "
+    "you make a <i>fake</i> payment. So, if you type the /start command, "
+    "you'll just get this screen again."
+)
+
+
+class FakePaymentScreen(Screen):
+    """The class implements FakePaymentScreen."""
+
+    description = FAKE_PAYMENT_SCREEN_DESCRIPTION
+
+    async def add_default_keyboard(self, _update, _context):
+        """Set up the keyboard for the screen."""
+        return [[
+            Button(
+                '❌ No',
+                PaymentScreen,
+                source_type=SourceTypes.MOVE_SOURCE_TYPE),
+            Button(
+                '✅ Yes',
+                self.handle_fake_payment,
+                source_type=SourceTypes.HANDLER_SOURCE_TYPE),
+        ]]
+
+    @ignore_permissions([PaywallPermission])
+    @register_button_handler
+    async def handle_fake_payment(self, update, context):
+        """Handle a button click and process a fake payment.
+
+        It is better to decorate all the handlers which are used
+        in a permission denied state.
+        """
+        user = update.effective_user
+        settings.PAID_USERS.append(user.id)
+
+        return await MainMenuScreen().move(update, context)
+
+    @ignore_permissions([PaywallPermission])
+    async def move(self, update, context, **kwargs):
+        """Switch to the screen re-rendering the previous message.
+
+        It is better to decorate all the handlers which are used
+        in a permission denied state.
+        """
+        return await super().move(update, context, **kwargs)
+
+
+class MainMenuScreen(StartMixin):
+    """The class implements MainMenuScreen that is only available after the payment."""
```

Comment:
`"""The class implements the MainMenuScreen, which is only available after a fake payment."""`